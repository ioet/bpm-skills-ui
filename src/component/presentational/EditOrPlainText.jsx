import React from 'react';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField/TextField';

const EditOrPlainText = (props) => {
  const {
    value, editId, categoryId, inputError, onCategoryEdit, onChange, name, label,
  } = props;

  return (
    (editId === categoryId)
      ? (
        <form onSubmit={(e) => {
          e.preventDefault();
          onCategoryEdit();
        }}
        >
          <TextField
            error={inputError}
            name={name}
            defaultValue={value}
            label={label}
            id="mui-theme-provider-input"
            onChange={
              (e) => {
                e.preventDefault();
                onChange(e);
              }
            }
          />
        </form>
      ) : (
        value
      )
  );
};

EditOrPlainText.defaultProps = {
  inputError: false,
  editId: undefined,
};

EditOrPlainText.propTypes = {
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  editId: PropTypes.string,
  categoryId: PropTypes.string.isRequired,
  inputError: PropTypes.bool,
  onCategoryEdit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
};

export default EditOrPlainText;
