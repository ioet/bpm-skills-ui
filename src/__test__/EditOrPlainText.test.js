import React from 'react';
import EditOrPlainText from '../component/presentational/EditOrPlainText';

describe('<EditOrPlainText />', () => {
  test('Component Renders correctly', () => {
    const appWrapper = shallow(<EditOrPlainText />);

    expect(toJson(appWrapper))
      .toMatchSnapshot();
  });
});
