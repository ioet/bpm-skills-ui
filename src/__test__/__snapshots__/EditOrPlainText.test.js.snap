// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditOrPlainText /> Component Renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <TextField
    error={false}
    id="mui-theme-provider-input"
    onChange={[Function]}
    required={false}
    select={false}
    variant="standard"
  />
</form>
`;
